server/leafnode.go

```diff
@@ -2215,6 +2222,11 @@ func (s *Server) leafNodeFinishConnectProcess(c *client) {
 
 	// Capture account before releasing lock
 	acc := c.acc
+	// cancel connectProcessTimeout
+	if !c.ping.tmr.Stop() {
+		<-c.ping.tmr.C
```

Comment:
I know that this is what Go asks us to do, but I have seen too many times this block. So if you do that, I would recommend having a select{} here. But honestly, I would just have something like:
```
clearTimer(&c.ping.tmr)
```
which takes care of checking if timer is nil or not and then Stop() and set it to nil.
If the timer can't be stopped because it is about to fire, it will do nothing since it will check the flag that you have set at top of function.

Reply:
done